### ISSUE TITLE
Armor and Inserts retrieved from insurance separate from one another

### ISSUE THREAD

**Issue by refringe** [sp-tarkov/server#554]
This issue was brought up in the #380-bleeding-edge-reports channel. Not by myself. I am just reporting the issue. 

Link to original message: https://discord.com/channels/875684761291599922/1203053345527300176/1203611038084370502

The issue happens when retrieving items from insurance.

When you insure armor, the inserts (the "plates" integrated into the armor) should be insured as well as the armor itself.(worth mentioning they wont have an insured icon as they are combined with the armor)
You can insure the front, back, and side plates all individually. If you insure the armor it automatically insures the front back and side plates (if any are inserted into the armor)

This user is saying that he got the inserts back from insurance, which shouldnt be possible because the inserts and armor are combined. However it is possible to get a front back or side plate and not the armor.

To reproduce:
Insure armor
Wait for insurance to come back
If the issue persists you will eventually get armor inserts and not the armor.

A few things to consider is:

Who you insure with (therapist or prapor)
How you insure (pre-raid screen or from the inventory)

TLDR:
Armor that cant be seperated from its inserts is being seperated and retrieved in insurance in parts.

I do not play SPT, I do not have it installed, I simply help out where I can because I enjoy the project and help my buddies who do not play live play SPT. This will explain why I reported for someone else and attached nothing. Hopefully you have plenty of information to go off of.

---

[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/401) by _**RubberBandyAndy**_

**Comment by refringe** [comment]
Rework of insurance should fix this: https://dev.sp-tarkov.com/SPT-AKI/Server/commit/a2668a96188949b344d378a7e1e9f0be3b0eb2e9

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/401#issuecomment-22660) by _**chomp**_

**Comment by refringe** [comment]
Profile added with issue, items in insurance. New zip should resolve this, just dropping here in case.

---
### Attachments
- [65bd5b020003c8282b3a4634.zip](https://dev.sp-tarkov.com/attachments/afeffc74-4230-45c0-a0c1-0b57a78186c0) (73 KiB)


---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/401#issuecomment-22704) by _**bandit**_

**Comment by refringe** [comment]
Found something:

```
Rolling deletion for "HighCom Trooper TFO body armor (MultiCam)" with THERAPIST - Return 85% - Roll: 95.91 - Status: Delete
Processing attachments for parent item: HighCom Trooper TFO body armor (MultiCam)
Child Item - Name: Monoclete level III PE ballistic plate, Max Price: 34306
Child Item - Name: Monoclete level III PE ballistic plate, Max Price: 34306
Rolling deletion with THERAPIST - Return 85% - Roll: 96.46 - Status: Delete
Rolling deletion with THERAPIST - Return 85% - Roll: 31.78 - Status: Keep
Number of successful rolls: 1
Marked for removal - Child Item: Monoclete level III PE ballistic plate, Max Price: 34306
```

Attachments for a parent item that has been marked as deleted should not be rolled for, they should just be deleted. It looks like the check I'm using to see if the item has children may not be working in the case of plates. Not sure if this is actually causing any issues, because I have yet to actually receive any individual plates back in my testing... even in this case.

I'll keep digging.

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/401#issuecomment-22860) by _**Refringe**_

**Comment by refringe** [comment]
I've just pushed [2aa0054833](https://dev.sp-tarkov.com/SPT-AKI/Server/commit/2aa0054833a030df52e301a86353cd01ceee288b) and [ebbc3e2029](https://dev.sp-tarkov.com/SPT-AKI/Server/commit/ebbc3e2029e2e85db462ad507e2cbb424e7f6a67) which include a number of improvements to the insurance package processing.

With these changes I've been able to confirm the following:

1. When armour is rolled for deletion, all of its plates are deleted as well.
2. When armour is not rolled for deletion, its plates are individually rolled for deletion.
3. When armour plates are rolled for deletion, the non-moddable plates are never rolled or deleted.

Attached is the profile that I used for testing.


---
### Attachments
- [65c2c6a1000556885070480a.json.zip](https://dev.sp-tarkov.com/attachments/bfcd8145-cd18-4a09-9e40-1240aa161ea4) (19 KiB)


---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/401#issuecomment-22862) by _**Refringe**_

**Comment by refringe** [comment]
Can confirm that armour is now returning with some missing plates still on it when the armor is kept and some plates are deleted. 

Haven't had an insurance roll mark the parent armour for deletion yet so will update when that happens

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/401#issuecomment-23246) by _**acidphantasm**_
